### ISSUE TITLE
Syncthing doesn't respect "run only on wifi" setting

### ISSUE THREAD

**Issue by klement** [syncthing/syncthing-android#1118]
My settings are:

Always run in background - ticked
Run only when charging - ticked
Run only on wifi - ticked
Restrict to certain wifi networks - none selected
Respect Android battery saving settings - ticked

The problem is: when I connect my phone to my car's charger, syncthing is running. Since this scenario happens mostly while roaming, it's extremely annoying (and quite costly).

### Version Information

    App Version: 0.10.10
    Syncthing Version: ?
    Android Version: Android 8.1.0

**Comment by Catfriend1** [comment]
Hi @klement ,
Syncthing Version of 0.10.10 is 0.14.47 btw. Can you please test if you can still reproduce this behaviour using the latest debug build https://build.syncthing.net/repository/download/SyncthingAndroid_Build/21858:id/debug/app-debug.apk ? (Careful - please backup your data first or use a different folder with a copy of your data. Please also don't forget to export your settings in the settings menu before switching over to the debug version and importing them.)
I've discovered various bugs related to the background "service" behaviour when using the run conditions and multiple (unwanted) starts of the syncthing service.

If you can still reproduce the issue, please try this grabbing some logs (e.g. syncthing log from the app, android log using Catlog app):
- Force close syncthing before "getting into the car" from android's app settings.
- Start the syncthing app - it will tell disabled with the phone in your hands.
- Let it be and switch to your home screen.
- Get into your car and plug the phone
Everything correct now?

I appreciate your feedback.

**Comment by Catfriend1** [comment]
Maybe a duplicate of https://github.com/syncthing/syncthing-android/issues/975 which seems to be a problem thats Android 8+ specific? I cannot reproduce this behaviour atm on Android 7.1.2 .

**Comment by AudriusButkevicius** [comment]
Feedback can live on the other ticket.

**Comment by AudriusButkevicius** [comment]
So given we've closed the other ticket as the 3 people there can't repro the issue, should we reopen this one and check if OP can provide logs/steps to repro?

**Comment by Catfriend1** [comment]
Right. @klement Can you please provide logs when running 0.10.10 and also running the test build (https://build.syncthing.net/repository/download/SyncthingAndroid_Build/21984:id/debug/app-debug.apk ) reporting back if the issue persists?

**Comment by klement** [comment]
I wanted to collect the log, but I can't reproduce the issue anymore. At
least not at home. I tried disabling the wifi and plugging the phone on
charger. Also tried with enabled wifi, but disconnected network. Both cases
worked as they should (syncthing not running). What changed today was that
I updated the software to the latest nexus 5x image, which google posted
yesterday (without wiping data). The only other differences while in the
car are 1.) bluetooth connected and 2.) both cases when I noticed it I was
roaming. Not sure if that could mess the syncthing logic.

**Comment by Catfriend1** [comment]
We'll reopen if you can reproduce it running the latest debug build at
https://build.syncthing.net/repository/download/SyncthingAndroid_Build/22103:id/debug/app-debug.apk
Please post the logs captured from start to occurence of the issue.

**Comment by klement** [comment]
Some more info:

I noticed again this behaviour. Situation was:

1. I wasn't roaming
2. There was no network coverage at the moment
3. I noticed that there is a question mark next to wifi symbol (aka "some open wifi available")

Maybe this can hint at what could be triggering the issue (the scan for open networks? does android briefly connect to such network? could this trigger syncthing to start?)

Next, I tried disabling wifi completely, since I was driving and trying to stop syncthing from sucking my mobile data. This, sadly did not do anything syncthing-wise. It kept running like nothing happened and to turn it off I had to go into syncthing settings and disable the runtime conditions and exit the app manually.
